### DISCUSSION TITLE
Tooltip only showing 2 of 50 dates

### DISCUSSION THREAD

**Discussion by libnine** [airbnb/visx#1076] [General]
This is probably something really silly, but my tooltip hover is only showing the first and last date of my dataset. Any idea what's going wrong?

Sandbox: https://codesandbox.io/s/bold-shockley-go3fb?file=/src/App.tsx

**Comment by williaster** [comment]
Hey @libnine 👋 . These things can be tricky. I took a look and it seems it's due to your data array ("`um`") not being sorted, which the [`d3` `bisector` helpers](https://github.com/d3/d3-array#bisectLeft) (used to find the nearest data point in your array) assume.

Here's a [modified sandbox](https://codesandbox.io/s/fragrant-dream-dycmq?file=/src/App.tsx:791-947) where I added a sort:

```ts
const um = dump
  .map((d: any) => {
    return { date: d[0], mean: d[8] };
  })
  .sort((a, b) => new Date(a.date).valueOf() - new Date(b.date).valueOf());
```

![image](https://user-images.githubusercontent.com/4496521/108924456-868db700-75ef-11eb-8aa5-441d36b39915.png)
